posts/20230126/index.qmd

```diff
@@ -0,0 +1,45 @@
+
+## Data leakage taxonomy
+* L1.1 No test set: This was shocking to us. How does the peer review process allow these papers to be published? Hopefully, in the future reviewers will be more demanding of ML practitioners. This is also the most common type of data leakage identified in Table 1. 
+* L1.2 Pre-processing on training and test set:  We felt that this was a loaded category. There are many different ways for this type of data leakage to come about. For example, during imputation, under/oversampling, and encoding. For clarity, we thought this category could have been subdivided.
+* L1.3 Feature selection on training and test set: This is the second most common type of data leakage found in Table 1. 
+* L1.4 Duplicates in dataset: This seems more like an error than a type of data leakage.
+* L2 Model features that aren't legitimate: We spent the most time discussing this category. We found it hard to think about proxy variables because they require a lot of domain knowledge. Instead, we like the definition of a feature the model would not have access to when making new predictions. 
+* L3.1 Temporal leakage: This is when future data is included in model training.
+* L3.2 Non-independence between training and test samples: This can be caused by resampling the same patient or geographic location. This type of leakage can also be due to the interpretation of the results. This occurs when there is a mismatch between the test distribution and the scientific claim. 
+* L3.3 Sampling bias in test distribution: This is choosing a non-representative subset for analysis.
+
+In general, categories L1.1-4 are clear cut and should be identifiable and avoidable. L2 is difficult because it depends on domain knowledge. Due to the nature of collaboration the researcher with this expert domain knowledge may not be the individual performing the ML analysis. Similarly, L3.1-3.3 are also difficult to detect and avoid if the researcher is using previously collected data and not designing an experiment and collecting data from scratch. In industry, data is often given to ML practitioners and there is no option to immediately collect more data or change the collection methodology. If this is the case, data leakage could be undetectable in many industry applications as ML practitioners are working with the best data at hand. In addition, there are limited strategies for identifying and mitigating these three types of data leakage. 
```

Comment:
"In industry, data is often given to ML practitioners and there is no option to immediately collect more data or change the collection methodology" -> Super important point